### PR TITLE
``nn.dense_mincut_pool`` Add temperature parameter to mincut pool operator

### PR DESCRIPTION
**Pull request by shhs29**
Hi,

I have added the temp argument to ``dense_mincut_pool`` operator with a default value of 1.0. I have not modified the corresponding tests. Kindly let me know if you want me to add tests with the new temp argument.

Thanks and Regards,
Shweta Ann Jacob

### CODE REVIEW

File: torch_geometric/nn/dense/mincut_pool.py [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
 from torch import Tensor
 
 
 def dense_mincut_pool(
-    x: Tensor,
-    adj: Tensor,
-    s: Tensor,
-    mask: Optional[Tensor] = None,
+    x: Tensor, adj: Tensor, s: Tensor, mask: Optional[Tensor] = None,
+    temp: Optional[Union[float, int]] = 1.0
```
Comment (EdisonLeeeee) [inline torch_geometric/nn/dense/mincut_pool.py:9]:
```suggestion
    temp: float = 1.0
```
+ `temp` cannot be `None`
+ `temp` is better to be `float`

Comment (shhs29) [reply torch_geometric/nn/dense/mincut_pool.py]: I have updated as per your suggestion.

File: torch_geometric/nn/dense/mincut_pool.py [hunk 2]
```diff
@@ -5,11 +5,8 @@
 
 
 def dense_mincut_pool(
-    x: Tensor,
-    adj: Tensor,
-    s: Tensor,
-    mask: Optional[Tensor] = None,
-) -> Tuple[Tensor, Tensor, Tensor, Tensor]:
+        x: Tensor, adj: Tensor, s: Tensor, mask: Optional[Tensor] = None,
+        temp: float = 1.0) -> Tuple[Tensor, Tensor, Tensor, Tensor]:
```
Comment (rusty1s) [inline torch_geometric/nn/dense/mincut_pool.py:9]:
```suggestion
        temp: float = 1.0, ) -> Tuple[Tensor, Tensor, Tensor, Tensor]:
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -52,6 +52,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added `BaseStorage.get()` functionality ([#5240](https://github.com/pyg-team/pytorch_geometric/pull/5240))
 - Added a test to confirm that `to_hetero` works with `SparseTensor` ([#5222](https://github.com/pyg-team/pytorch_geometric/pull/5222))
 ### Changed
+- Support temperature with `float` or `int` in `dense_mincut_pool` ([#5908](https://github.com/pyg-team/pytorch_geometric/pull/5908))
```
Comment (rusty1s) [inline CHANGELOG.md:55]:
```suggestion
- Support temperature value in `dense_mincut_pool` ([#5908](https://github.com/pyg-team/pytorch_geometric/pull/5908))
```

File: torch_geometric/nn/dense/mincut_pool.py [hunk 4]
```diff
@@ -65,7 +64,7 @@ def dense_mincut_pool(
 
     (batch_size, num_nodes, _), k = x.size(), s.size(-1)
 
-    s = torch.softmax(s, dim=-1)
+    s = torch.softmax(s / temp, dim=-1)
```
Comment (rusty1s) [inline torch_geometric/nn/dense/mincut_pool.py:67]:
```suggestion
    s = torch.softmax(s / temp if temp != 1.0 else s, dim=-1)
```

**Review by rusty1s — APPROVED** [review]

**Pull request merged**